include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -235,10 +278,10 @@ class fm_index
         // uint8_t largest_char = 0;
         sdsl::int_vector<8> tmp_text(text.size());
```

Comment:
do you static_assert somewhere that the alphabet size can b 256 at most?

Reply:
For the fm-index:
`std::Same<underlying_rank_t<innermost_value_type_t<text_t>>, uint8_t>`

... which also has the problems with the rank_type

Reply:
I would instead check that the size is <=256 with a comment that says that 256 and 255 are not really legally, but have a workaround that throws.